### ISSUE TITLE
[data grid] Tooltip on row checkbox

### ISSUE THREAD

**Issue by nickbar86b** [mui/mui-x#6088]
### Order ID 💳

50436

### The problem in depth 🔍

We are trying to add a tooltip on our checkboxes based on row info for DataGridPro.
As far as I understand there are 2 ways to do this, either override the column or override the component it self.
In the first case we could not just simply use 'Checkbox' cause it would break select all functionality so we ended up doing something like
```
const columns: GridColDef<ModelObject>[] = [
    {
      ...GRID_CHECKBOX_SELECTION_COL_DEF,
      renderCell: (params) => {
        if (existing.includes(params.row.key)) {
          return (
            <Tooltip placement={'bottom'} title={'Already added to story'}>
              <div>
                <GridCellCheckboxRenderer {...params} />
              </div>
            </Tooltip>
          );
        } else {
          return <GridCellCheckboxRenderer {...params} />;
        }
      },
      width: 100,
      type: 'singleSelect',
    },
```
But it feels wrong to use GridCellCheckboxRenderer to achieve this.

Overriding  checkbox through components props is not giving me the row info.
What is the best way to do this?

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by m4theushw** [comment]
What you're doing is the right way. If you pass a component to `rootProps.components.BaseCheckbox` it won't be used only for the selection checkbox, but also for the Select All checkbox and the component rendered when editing booleans. Using this approach doesn't give the row info because the checkbox can be rendered in many places. Providing a custom `renderCell` is recommended because it gives the row data and changes only the selection checkbox.

**Comment by nickbar86b** [comment]
Yes but I find it unnecessary complicated to use MUI Checkbox. As far as I understand GridCellCheckboxRenderer is an internal component(founded by just reading the code and not in any documentation page.) and probably is not the best way to handle this use case. Is there an exported component I can use which I can use to simply apply the tooltip(similarly to what I do above) and keep all the rest of the checkbox functionalities the same? Or is it fine  to use GridCellCheckboxRenderer

**Comment by nickbar86b** [comment]
Effectively I am finding hard to understand the API "my" Checkbox should implement to work seamlessly with all the rest of DataGrid Functionality

**Comment by m4theushw** [comment]
`GridCellCheckboxRenderer` is not an internal component. If it's exported you can use it as stable. We just don't have documentation for it, yet, but we want to document everything that is exported.

**Comment by nickbar86b** [comment]
@m4theushw I think I found a bug. 
Given the code bellow 
`const columns: GridColDef<ModelObject>[] = [
    {
      ...GRID_CHECKBOX_SELECTION_COL_DEF,
      renderCell: (params) => {
        if (existing.includes(params.row.key)) {
          return (
            <Tooltip
              placement={'left'}
              title={'Already added'}
              followCursor={false}
            >
              <div>
              <GridCellCheckboxRenderer {...params} />
              </div>
            </Tooltip>
          );
        } else {
          return (
            <Tooltip
              placement={'left'}
              title={'Select transaction'}
              followCursor={false}
            >
              <GridCellCheckboxRenderer {...params} />
            </Tooltip>
          );
        }
      },
      width: 100,
      type: 'singleSelect',
    },`
When you place your cursor on the checkbox columns and start scrolling down slowly, the tooltip appears but as you continue scrolling down the scrolling behaviour become flunky and the it goes up and down while the tooltip is being closed.

**Comment by nickbar86b** [comment]
and also in file GridHeaderCheckbox.d.ts
`declare const GridHeaderCheckbox: React.ForwardRefExoticComponent<GridColumnHeaderParams<any, import("../../models/gridRows").GridValidRowModel, any> & React.RefAttributes<HTMLInputElement>>;`
should be 
`declare const GridHeaderCheckbox: React.ForwardRefExoticComponent<R extends GridColumnHeaderParams<any, import("../../models/gridRows").GridValidRowModel, any> & React.RefAttributes<HTMLInputElement>>;`

**Comment by nickbar86b** [comment]
@m4theushw kind reminder

**Comment by michelengelen** [comment]
The guidance above still stands and is the supported approach: spread `GRID_CHECKBOX_SELECTION_COL_DEF`, provide a custom `renderCell`, and wrap `GridCellCheckboxRenderer` in your `Tooltip`. `GridCellCheckboxRenderer` remains a public export, so building on it is safe; the `baseCheckbox` slot is intentionally not row-aware because it also renders the header checkbox and boolean editors. The scrolling quirk you hit was the tooltip portal, solved with `slotProps={{ popper: { disablePortal: true } }}` on the Tooltip.

Closing as answered. If the checkbox renderer loses functionality inside a wrapper on the latest version, please open a new issue.
